1 html

Ruslan1930 opened the pull request:
В 1 задании не уверен, что правильно выполнил добавление кнопки "Поиск"
Во 2-м задании были сомнения с необходимостью созданию 2-х форм, но, по сути кнопки "Войти" и Зарегистрироваться должны направить на разные формы (страницы)

Ruslan1930 committed:
Первый коммит с дз blog

Ruslan1930 committed:
Подкорректировал код

Ruslan1930 committed:
2-blog

GitEsense commented on 1-Blog/blog.html:
```diff
@@ -0,0 +1,45 @@
+<body>
+    <header>
+    <nav>
+        <button>В</button>
+        <form action="#">
+            <label for="search"></label>
+            <input type="text" name="search" id="search" placeholder="Поиск">
+        </form>
+        <ul>
+            <li><a href="#">Мой блог</a></li>
+            <li><a href="#">HTML и CSS</a></li>
+            <li><a href="#">Backend</a></li>
+            <li><a href="#">Поиск</a></li>
+        </ul>
+    </nav>
```
Лучше всего вынести "логотип" за навигационный блок, так же это будет ссылкой (не формой), форма поиска идёт по макету "после", конечно можно и вырывать из контекста в перспективе, но лучше оформить
```html
<a><span>В</span>Мой блог</a>
<nav>
    ....
<div>
   <label>
   <input>
```
Суть логики какая 
1. Логотип первым элементом (ссылка)
2. Навигационное меню (список ссылок)
3. инпут (форма поиска)

GitEsense commented on 1-Blog/blog.html:
```diff
@@ -0,0 +1,45 @@
+            <p>Наталья Васильева</p>
+            <p>Разработчик</p>
+        </section>
+    <section>
+        <img src="./image/4f3f91ff718dae25d01f84692996a2cd.png" alt="">
+    </section>
```
Можно отдельным section, можно div, но лучше воспользоваться div (обёртка контента), т.к. section гласит, что у нас идут элементы, неразрывно связаны друг с другом (единый контент), а в данном случае один элемент), в прочем и обернуть в секцию - не критично)

GitEsense commented on 2-Form/form.html:
```diff
@@ -0,0 +1,35 @@
+</head>
+<body>
+    <main>
+        <h1>Регистрация</h1>
+        <span>Уже есть аккаунт?</span>
+        <form action="#">
+            <button>Войти</button>
+        </form>
+        <h2>Моя компания</h2>
+        <form action="###">
+           <div>
+                <label for="name"></label>
+                <input type="name" id="name" name="name" placeholder="Имя">
+                <label for="surname"></label>
+                <input type="text" placeholder="Фамилия" id="surname" name="surname">
+            </div> 
+            <div>
+                <label for="email"></label>
+                <input type="email" placeholder="E-mail" id="email" name="email" required>
+            </div>
+            <button>Зарегестрироваться</button>
+        </form>
+        <footer>
+            <span>Я прочитал и принимаю</span>
+            <a href="##">Условия пользования</a>
+        </footer>
+    </main>
```
Это не разные формы - это одна форма с целым списком инпутов (если в данном случае нажать "Войти" - то будет отправлена пустая форма - это не совсем то поведение, Вход сделан как ссылка на "логин - форму")
Так же h1, span лучше перенести в саму форму, т.к. является "заголовком формы", следовательно "Войти" - это не кнопка - это ссылка.

Моя компания - это тоже инпут, просто другого оформления, отличного от других

Ruslan1930 replied:
Вот и у меня были сомнения, что в случае создания единой формы у нас кнопка "Войти" будет направлять на один и тот же action,  моя ошибка была в том, что я сделал кнопку, а нужно было ссылку (теперь все ясно).

GitEsense commented on 2-Form/form.html:
```diff
@@ -0,0 +1,35 @@
+            </div>
+            <button>Зарегестрироваться</button>
+        </form>
+        <footer>
+            <span>Я прочитал и принимаю</span>
+            <a href="##">Условия пользования</a>
+        </footer>
```
Это футер формы - отлично, пусть им и будет, но у Вас пропущен "инпут" типа checkbox

Ruslan1930 replied:
Внес исправления по всем замечаниям

GitEsense reviewed (changes requested):
Обратите внимание на комментарии, пройдитесь по "занятиям" которые не понятны, если возникло недопонимание, будут вопросы- не стесняйтесь задавать в чате, после внесения изменений, обязательно коммит и отправить на повторную проверку на платформе

Ruslan1930 committed:
Исправил

Ruslan1930 committed:
Исправил. (Также подумал, что для h1,span,a  нужно создать div

GitEsense commented on 2-Form/form.html:
```diff
@@ -0,0 +1,40 @@
+            </div>
+            <button>Зарегестрироваться</button>
+        </form>
+        <footer>
+            <label for="checked"></label>
+            <input type="checkbox" id="checked">
+            <span>Я прочитал и принимаю</span>
+            <a href="##">Условия пользования</a>
+        </footer>
```
Вот эта часть так же - форма , это ещё один инпут, который при отправке формы передаст значение "true", но его не будет, т.к. во-первых не входит в форму, во-вторых, отсутствует передаваемый параметр "name", положите этот блок в форму и дайте инпуту "имя", так же span + a  должны быть частью лейбла, иначе это будет просто текст, а как известно - лейбл - это дополнительный элемент, связанный с инпутом, при клике на который (в данном случае текст) - будет переводить чекбокс в смену состояний (checked и обратно), чтобы сейчас сделать отметку - необходимо нажать на сам квадратик.

Ruslan1930 replied:
Понял. Исправляю

GitEsense reviewed (commented):
По формам обратите внимание, внесите корректировку и уже будем "апрувить", рекомендую ещё раз ознакомиться о связке лейбл \ инпут и поведении формы, а так же как передаются значения (и про атрибуты)

Ruslan1930 committed:
Исправил

GitEsense reviewed (approved):
Вот теперь отлично, можно мерджить)

Ruslan1930 committed:
Подкорректировал

Добавил стрелку к кнопке "Зарегестрироваться" стрелку и подключил css

Ruslan1930 committed:
ДЗ 7.9 Основы CSS

Ruslan1930 committed:
Добавил коммит для код ревью ДЗ 12.9

Постарался сделать максимально приблеженно, как на фигме, знаю, что есть ошибки, но нужна обратная связь. т.к. идей для улучшения кода пока нет

GitEsense commented on 1-Blog/blog.html:
```diff
@@ -0,0 +1,51 @@
+            <form action="#">
+                <label for="search"><img src="../1-Blog/image/search.svg" alt=""></label>
+                <input type="text" name="search" id="search" placeholder="Поиск" class="poisk" >
+                <!-- Не совсем понимаю как вставить картинку так, чтобы она была внутри инпута -->
```
Чтобы картинку вставить внутрь инпута - можно прибегнуть к нескольким способам:
1. Убрать изображение из лейбл и сделать такой стиль
```css
form {
   position: relative;
}
label::before {
   position: absolute;
   content: url('../1-Blog/image/search.svg');
   top: 8px;
   left: 11px;
}
```

2. или такой вариант
```css
form {
   position: relative;
}

label > img {
   position: absolute;
   content: url('../1-Blog/image/search.svg');
   top: 8px;
   left: 11px;
}
```

GitEsense commented on 1-Blog/styles/style.css:
```diff
@@ -0,0 +1,112 @@
+* {
+    background-color: #374151;
+    box-sizing: border-box;
+}
```
Вот тут будет очень ошибочно указывать background-color, т.к. класс * применяется КО ВСЕМ ЭЛЕМЕНТАМ в документе, т.е. каждый элемент имеет такой бекграунд, лучше бг перенести в body

GitEsense commented on 1-Blog/styles/style.css:
```diff
@@ -0,0 +1,112 @@
+    /* Почему-то он перечеркнул мне фонт-вэйт и пришлось заново перезадавать, хотя в Body стоит font-weight 400 */
+}
+h1 {
+    margin: 0 457px 30px 457px;
```
Такие марджины необходимо убрать, т.к. позиционируем мы не относительно конкретного размера, а динамично (и если разрешение будет 1080px мы этого блока и вовсе можем не увидеть и будет лишний скрол

Лучше ограничить размер body , например max-width: 1080px, сделать margin: 0 auto (разместить в центре) и уже внутри этого body все блоки будут идти "сами по себе", если нам надо в центр например section - делаем то же самое (если надо ограничить layout

Например блоки текста (нижние)  имеют ширину 708px (т.е. можно сделать max-width: 708px; margin: 0 auto) и уже все дочерние параграфы будут такого размера

Например стилизовать можно section так
```css
section {
   max-width: 525px;
   text-align: center;
   margin: 0 auto
}
.info {
   max-width: 708px;
   text-align: start;
}

div.image {
   max-width: 896px;
   margin: 0 auto;
}
```

GitEsense commented on 1-Blog/blog.html:
```diff
@@ -0,0 +1,51 @@
+                <p class="profession">Разработчик</p>
+            </div>
+        </section>
+        <div>
+            <img src="./image/4f3f91ff718dae25d01f84692996a2cd.png" alt="" class="pc">
+        </div>
```
накинуть класс image на div

GitEsense commented on 2-Form/form.html:
```diff
@@ -0,0 +1,39 @@
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Document</title>
+    <link rel="stylesheet" href="/css/2-css.css">
```
Необходимо поправить ссылку или положить верно файл стиля, лучше его просто назвать "main.css" или "style.css"

GitEsense commented on css/2-css.css:
```diff
@@ -0,0 +1,66 @@
+body {
+    background-color: #F39494;
+}
```
```css
* {
    box-sizing: border-box;
}
body {
    background: linear-gradient(
            225deg,
            #fdc5c5 25.45%,
            rgba(242, 152, 222, 0) 100%
        ),
        linear-gradient(225deg, #ffa5a5 33.09%, rgba(255, 255, 255, 0) 82.34%),
        #f39494;
    margin: 0 auto;
    max-width: 1440px;
    height: 100%
}
html {
   height: 100%
}
```
Это бекграунд фона

Так же надо бы добавить

GitEsense reviewed (changes requested):
Пройдитесь ещё раз по позиционированию, внутренним отступам, центровке, position absolute и т.п.
Посмотрите поведение блочных элементов и каскад стилей (как можно наложить наследование и переиспользовать стили), примеры я Вам привёл (считай готовое решение), но нужно, чтобы Вы поняли сами и попробуйте переделать по-своему, дать классы нужным блокам, обернуть и поиграть с разными вариантами

Ruslan1930 committed:
Исправления по ДЗ

Попробовал исправить согласно Вашим указаниям, но не стал добавлять text-align, так как не дошел до него и не знаю как правильно им пользоваться. также не применял градиентную заливку, которую Вы мне отправили, так как отображается совсем другой цвет

GitEsense commented on 1-Blog/styles/2style.css:
```diff
@@ -0,0 +1,188 @@
+    position: absolute;
+    content: url(../image/search.svg);
+    top: 8px;
+    left: 11px;
```
По макету у нас
```css
top: 14px;
left: 18px;
```

GitEsense commented on 1-Blog/styles/2style.css:
```diff
@@ -0,0 +1,188 @@
+    content: url(../image/search.svg);
+    top: 8px;
+    left: 11px;
+}
```
Дублирование кода... внимательнее, дубли надо убрать

GitEsense commented on 1-Blog/styles/2style.css:
```diff
@@ -0,0 +1,188 @@
+    font-size: 24px;
+    font-family: 'Roboto';
+    text-align: center;
+}
```
Вот поэтому и перечеркнул - Вы переопределяете его дальше, h1 несколько раз переопределяется

GitEsense commented on 1-Blog/styles/style.css:
```diff
@@ -1,112 +1,160 @@
+    border-radius: 100px;
+    background: rgba(31, 41, 55, 0.6);
+    padding: 10px 30px;
     color: #FFFFFF;
```
Так же проверьтесь на дубли и соберите воедино...
по макету 
```css
padding: 12px 11px 9px 49px;
```
Внимательнее с "деталями"

GitEsense commented on 1-Blog/styles/style.css:
```diff
@@ -1,112 +1,160 @@
-h1 {
-    margin: 0 457px 30px 457px;
+ul:nth-child(1) {
+   margin-right: 52px;
```
li:nth-child(1 \2 \ etc.), мы применяем к li, а не ul

GitEsense commented on 1-Blog/styles/style.css:
```diff
@@ -1,112 +1,160 @@
+}
+.heroB {
+    width: 35px;
+    height: 35px;
```
не будут работать, не является блоком, можно сделать 
display: inline-block; тогда будет работать
но лучше реализовать заливку как 
```css
padding: 6px 12px;
```
числа "условные"

GitEsense commented on 1-Blog/styles/style.css:
```diff
@@ -0,0 +1,160 @@
+    line-height: 30px;
+    font-size: 16px;
+    font-weight: normal;
+    margin: 0 365px 103px 365px;
```
Если хотите по центру выствить, то сделайте margin: 0 auto (а не фиксированные пиксели), т.к. при изменении размера окна - отступы могут поехать и блок уедет

GitEsense reviewed (changes requested):
Обратите внимание на комментарии, можно много где упростить, не задавая лишних высот или размеров ширины, чтобы не дублировать, так же дубл стилей текста можно вынести отдельно, комментарии относятся не к конкретному месту, а ко всем стилям - обратите внимание

Ruslan1930 committed:
Работа над ощибками ДЗ 12.9

Внес коррективы, также удалил файл 2style.css т.к. он ввел в заблуждение
Рабочим файлом является style.css

GitEsense commented on 1-Blog/styles/style.css:
```diff
@@ -31,13 +29,13 @@ header > a {
-ul:nth-child(2) {
+li:nth-child(2) {
     margin-right: 39px;
 }
```
А чтобы это заработало, Вам необходимо выставить "display: inline-block", иначе они как блочные элементы занимают всё свободное пространство, а inline позволит сделать "по размеру контента", то же можно выставить и на nav, а для logo выставить margin-right: 77px;
search прижимать к краю не нужно (но если хотите - можете воспользоваться абсолютным позиционированием

GitEsense commented on 1-Blog/styles/style.css:
```diff
@@ -61,37 +59,38 @@ form {
-    margin: 0 586px 100px 640px;
+    margin-bottom: 98px;
+    padding: 0;
+    /* Отступ не делается, хотя в макете показано, что отступ 28px */
```
а вот тут отступ не делается, потому что у Вас блок уходит сильно ниже, мешает Вам высота (height:53px), т.к. у Вас блочное поведение, то параграфы перепрыгивают, решить Вы можете выносом img из div.user и добаить в .user display: inline-block;

GitEsense commented on 1-Blog/styles/style.css:
```diff
@@ -61,37 +59,38 @@ form {
+    /* Отступ не делается, хотя в макете показано, что отступ 28px */
 }
 .userPhoto {
     width: 50px;
```
Лучше воспользоваться max-width:50px и картинка, если Вы посмотрите явно не соответствует макету, я уже говорил ранее - либо взять "круглях", либо стилизовать в соответствие)

GitEsense reviewed (commented):
Форму по прежнему не привели в порядок)
по фактическим размерам высоты я уже озвучивал, сейчас Вы столкнулись с последствиями (в блоке .user)
посмотрите комментарии (в том числе и предыдущие)

Ruslan1930 committed:
Исправил ДЗ 12.9

у меня больше не осталось идей, как можно еще оптимизировать:)

Ruslan1930 committed:
Внес исправления

GitEsense reviewed (changes requested):
По блогу - проверьте шрифты (цвета, размеры)
Вы можете отпозиционировать блоки first section через text-align: center;
Картинку вытащить из div (основную), выставить стили на неё и она так же получит "text-align", тогда margin будет не нужен, в противном случае надо будет вешать на div overflow: hidden


Так же форма явно не похожа на макетную, её надо доработать (не просто налепить стили, форму Вы можете сделать на 95% согласно макету и её так и надо сделать

Ruslan1930 committed:
Внес исправления в ДЗ

GitEsense reviewed (changes requested):
По форме (позиционированию) Вы можете сделать:
1. на форму (саму форму) сделать padding: 72px 95px 91px 93px;
![chrome_nxo4JOkVU3](https://github.com/Ruslan1930/DZ-demo/assets/101252030/14f8d070-587f-4475-b2dd-2aeb879b1d96)
Тогда все элементы внутри уже будут по этому отступу выставляться ( и не нужны вечные margin: 0 auto;

2. Указать размеры (394px или в перспективе узнаете о процентах, т.е. 100%) - растянет на максимальный размер (в нашем случае это 394px
3. Кнопка у нас имеет padding: 9px 12px  (остальное - растягивает по размеру блока), нужен width (394px) или display: block и width: 100%;
4. Чтобы форму поставить "в центр экрана" - нужен position: absolute (как вариант)
5. В кнопке иконка стрелки у Вас стоит через absolute, но не совсем верно (ознакомьтесь повторно если необходимо)
```css
    position: absolute;
    content: url(../images/Arrow\ 1.svg);
    padding-left: 56px;
}
```
заменить на 
```css
   button {
   position: relative;
}
button::after {
    position: absolute;
    content: url(../images/Arrow\ 1.svg);
    right: 24px;
}
```

Ruslan1930 committed:
Задание по макету с формой

Внес исправление в 2-Form

GitEsense reviewed (approved):
Сильно придираться не буду, тут много над чем можно поработать для приведения 1 к 1, но это мы будем делать потом, по блочной вёрстке +- у Вас понимание есть, однако, отступы, чтение макета, радиусы и размеры всё-таки надо будет повторить прежде чем идти дальше, Вам предстоит на ДЗ "3-flex" кропотливая работа, раз не реализуем в рамках этого ДЗ - будем доводить до 100% соответствия уже на "флекс" ДЗ
